gnucash/gnome/gnc-budget-view.c

```diff
@@ -815,16 +817,29 @@ budget_accum_helper(Account* account, gpointer data)
 {
     BudgetAccumulationInfo* info = (BudgetAccumulationInfo*)data;
     gnc_numeric numeric;
+    gnc_commodity *currency;
+
+    currency = gnc_account_get_currency_or_parent(account);
 
     if (gnc_budget_is_account_period_value_set(info->budget, account, info->period_num))
     {
-        numeric = gnc_budget_get_account_period_value(info->budget, account, info->period_num);
-        info->total = gnc_numeric_add(info->total, numeric, GNC_DENOM_AUTO, GNC_HOW_DENOM_LCD);
+        numeric = gnc_budget_get_account_period_value(info->budget, account,
+                                                      info->period_num);
+        numeric = gnc_pricedb_convert_balance_nearest_price_t64(
+                    info->pdb, numeric, currency, info->total_currency,
+                    gnc_budget_get_period_start_date(info->budget, info->period_num));
```

Comment:
This isn't risk-free. If the pricedb doesn't have a conversion either between the two commodities or between each of the two and a common third commodity then it will return 0. You might want to test for that and raise an error.

Reply:
OK, @jralls  but I used the same logic that's in "gnc_ui_accounts_recurse" from the main summary bar and also in xaccAccountConvertBalanceToCurrencyAsOfDate and xaccAccountGetBalanceAsOfDateInCurrency.
None of those do any checking or error raising. The later is the one providing the balances for the main Account page.
So the behavior in Budget would be the same than in the main page, as the commodities that need conversion are exactly the same, as I'm adding and converting over the same account tree.
If something needs to be done, it should be done consistently everywhere, not only in budget.

Reply:
OK, though one could argue that zeroing out a budget item for the lack of an exchange rate would impact the user rather more than would a total on the summary bar.